haystack/core/pipeline/base.py

```diff
@@ -710,9 +704,42 @@ def show(
         :param timeout:
             Timeout in seconds for the request to the Mermaid server.
 
+        :param super_component_expansion:
+            If set to True and the pipeline contains SuperComponents the diagram will show the internal structure of
+            super-components as if they were components part of the pipeline instead of a "black-box".
+            Otherwise, only the super-component itself will be displayed.
+
         :raises PipelineDrawingError:
             If the function is called outside of a Jupyter notebook or if there is an issue with rendering.
         """
+        import warnings
+
+        warnings.warn(
```

Comment:
Currently, we would show this warning even if there are no positional arguments. Can we show the warning only if there are any positional arguments? For example making use of [inspect.getargvalues ](https://docs.python.org/3/library/inspect.html#inspect.getargvalues) or with something similar to [this](https://stackoverflow.com/a/68432663)?

Reply:
I would suggest to show it either way - just calling the function should be a enough to trigger the warning to the user that it will have some breaking  changes in the future.

I don't think it's noisy if we show it even if he/she doesn't call it with positional arguments.

Reply:
I don't think it's good to shadow this from the user, especially if the user calls the function for whatever use he/she might have.

Anyway, I will code this only to trigger the warning in case the user calls the draw() or show() function using positional arguments.